src/electron/main.ts

```diff
@@ -305,14 +305,34 @@ const userStore = new ElectronStore();
 					return;
 				}
 
-				const analysis = await Analyzer.analyze(connection.path, {
-					getGobalEnumOptionId: (patternId, contextId) =>
-						library.assignEnumOptionId(patternId, contextId),
-					getGlobalPatternId: contextId => library.assignPatternId(contextId),
-					getGlobalPropertyId: (patternId, contextId) =>
-						library.assignPropertyId(patternId, contextId),
-					getGlobalSlotId: (patternId, contextId) => library.assignSlotId(patternId, contextId)
-				});
+				let analysis;
```

Comment:
Let's avoid the `let` binding by defining a `getAnalysis` function that returns a `AnalysisResult`: 

```ts
type AnalysisResult =
  | AnalysisError
  | AnalysisSuccess;

enum AnalysisState {
   Error,
   Success
}

interface AnalysisError {
   state: AnalysisState.Error;
   error: Error;
}

interface AnalysisSuccess {
   state: AnalysisState.Success;
   analysis: Analysis;
}

const getAnalysis = async (path: string; context: AnalysisContext): AnalysisResult => {
  try {
    return { state: AnalysisState.Sucess, analysis: Analyzer.analyze(path, context) };
  } catch (error) {
    return { state: AnalysisState.Error, error };
  }
};
```